lib/sort_by_length.rb

```diff
@@ -1,7 +1,12 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2) n is the split_sentence array which gets iterated over until there are no more elements left
+# Space complexity: O(n) n is for the sorted array which takes up additional space
 def sort_by_length(my_sentence)
```

Comment:
This is a selection sort, but with O(n) space complexity.  Can you do it with O(1) space complexity?